### PR TITLE
feature: Publish cohort SDK NPSs with bundled rust src

### PR DESCRIPTION
**Pull request by fmarek-kindred**

### CODE REVIEW

File: .github/workflows/npm.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+name: cohort_sdk_js
```
Comment (fmarek-kindred) [inline .github/workflows/npm.yml:1]: This workflow does not work and it is not activated

File: .github/workflows/npm_publish_light.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
+name: Build and Publish Cohort SDK NPMs
+on:
+    #workflow_call:
+    push:
```
Comment (fmarek-kindred) [inline .github/workflows/npm_publish_light.yml:4]: Now triggered on push, but after PR will change to be on manual button click.

File: packages/cohort_sdk_js/npm/darwin-arm64/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (fmarek-kindred) [inline packages/cohort_sdk_js/npm/darwin-arm64/package.json:1]: files in npm/ are not used.

File: scripts/github-actions-ci/set-npm-dev-version.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+#   $1 - Required. The package-name of node module
+#   $2 - Optional. The parameter name under which the resulting value will be exported into
+#        $GITHUB_OUTPUT
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+selectNodeName=$1
+outputParamName=$2
+
+if [ "$selectNodeName" == "" ]; then
+    echo "Missing the furst parameter - npm package name"
```
Comment (gk-kindred) [inline scripts/github-actions-ci/set-npm-dev-version.sh:19]: small typo `furst` -> `first`

File: packages/cohort_sdk_js/package.json [hunk 5]
```diff
@@ -25,14 +28,23 @@
...
+    "index.js",
+    "cohort_sdk_js*",
+    "scripts/postinstall.sh"
+  ],
   "scripts": {
-    "artifacts": "napi artifacts",
+    "artifacts": "napi artifacts -d .",
     "build": "napi build --platform --release",
     "build:debug": "napi build --platform",
     "test": "ava",
```
Comment (gk-kindred) [inline packages/cohort_sdk_js/package.json:42]: Hmm, interesting, haven't seen `ava` in years. 

Just wanted to know why you why choose `ava` over some of the other popular testing libraries like `jest`, `mocha`

File: .github/workflows/npm_publish_light.yml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+        name: Publish SDK NPS with bundled Rust for node@16
+        runs-on: ${{ matrix.settings.host }}
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Setup node
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 16
```
Comment (gk-kindred) [inline .github/workflows/npm_publish_light.yml:29]: Probably use `16` and `18`

File: .github/workflows/npm_publish_light.yml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+        strategy:
+            fail-fast: false
+            matrix:
+                settings:
+                  - host: macos-latest
+                    dirSdkJs: packages/cohort_sdk_js
+                    dirSdkJsClient: cohort_sdk_client
+                    dirCohortInitiator: cohort_banking_initiator_js
+                    dirCohortReplicator: cohort_banking_replicator_js
+                    customiseHostCommand: echo "No customizations needed"
```
Comment (gk-kindred) [inline .github/workflows/npm_publish_light.yml:19]: I probably wouldn't pass custom commands over a variable and inject it. 

Two reasons:- 
- This feels a bit unsafe to me. 
- Also when I look at the actions it makes it harder to interpret the workflow when custom commands come from variables. (I referred to `npm.yml` file and that is why this comment)

File: cohort_banking_initiator_js/README.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
+# Cohort Initiator JS App
```
Comment (gk-kindred) [inline cohort_banking_initiator_js/README.md:1]: Nice 👍 👍  Thanks for writing this

File: packages/cohort_sdk_js/package.json (hunk 5 above)
Comment (fmarek-kindred) [reply packages/cohort_sdk_js/package.json:42]: I did not, it was copy from scylla. In fact no testing is done in this repo. Let me remove it.

File: .github/workflows/npm_publish_light.yml (hunk 6 above)
Comment (fmarek-kindred) [reply .github/workflows/npm_publish_light.yml:29]: We have ticket to do testing with node 18. If we are going to support both then the whole workflow should be different in the way that it should be publishing two artefacts, one for v16 and one for v18. I prefer to have that work done as separate ticket (if we need to do it)

**Review by gk-kindred — APPROVED** [review]
Thanks for updating 👍

**Pull request merged**